### ISSUE TITLE
Duplicate entries across pages when sorting in Strapi-PostgreSQL integration

### ISSUE THREAD

**Issue by shinchannn** [strapi/strapi#21393]
### Required System information

- Node.js version: v18.20.2
- NPM version: 10.5.0
- Strapi version: 5.0.0
- Database: PostgreSQL
- Operating system: macOS Sequoia Version 15.0
- Is your project Javascript or Typescript: Javascript

### Describe the bug

When using Strapi with `PostgreSQL`, the Content Manager's pagination malfunctions when there are ties in the sorted field. Specifically, when sorting by a field with duplicate values (in this case, `county`), some items erroneously appear on multiple pages. This issue was first encountered in Strapi v4.11.4 and has been reproduced in v5.0.0, indicating it's not fixed in the latest release. Notably, the bug doesn't manifest when using SQLite (in which it seems `id` are used as tie breaker) , suggesting it might be related to PostgreSQL's handling of ties. MySQL hasn't been tested, it may use different tie-breaking mechanisms. This pagination inconsistency disrupts the user experience and data presentation in the Content Manager.

### Steps to reproduce the behavior

1. Create a new Strapi project using the CLI command `npx create-strapi-app`.
2. During setup, opt out of the default SQLite database. Instead, configure a PostgreSQL connection. (DigitalOcean hosted database for this reproduction).
3. Start the newly created project.
4. Create a new content type called **School**, with two fields:`name` and `county`.
5. Populate the "School" content type with the provided test data (provided at the bottom).
6. Set items per page to 10 and then, sort by `county`.

### Expected behavior

You should see some item showed in one page pop up on other pages. 

### Screenshots

Here in my case:
![bug](https://github.com/user-attachments/assets/308d6ab8-f6ad-406f-ad00-ee6badb4bad0)

### Code snippets

If applicable, add code samples to help explain your problem.

### Additional context

The public API, which leverages the same underlying functionality found in the entity manager, is equally affected by this pagination anomaly.

After investigation, I found executed SQL queries in entity manager look like:
- select `t0`.*, `t0`.`created_by_id`, `t0`.`id`, `t0`.`updated_by_id`, `t0`.`document_id` from `schools` as `t0` where (`t0`.`published_at` is null and `t0`.`published_at` is null) order by `t0`.`county` ASC limit 10
- select `t0`.`id`, `t0`.`document_id`, `t0`.`locale`, `t0`.`updated_at`, `t0`.`created_at`, `t0`.`published_at` from `schools` as `t0` where (`t0`.`published_at` is not null and `t0`.`document_id` in ('jdakhr8fciu86l8w09z57hu2', 'c38akhlpgdse5sqsxfodaldw', 'ueoidezy4rve3kho04v2rxt1', 'voxdca0nthipmj2hio57v7hc', 'jhylv48ar2n107dmhcxqg0mg', 'nlfffmb803gvrpihqd70k0o7', 'qv7cgqi7vjwfkpyq23yn3u8t', 'hgs0aw8j8j4oxr69xfm82pe1', 'u2rbvp8cyjsfu5jrp64ur706', 'pqsqhuktthkb60u2g5o4ef03') and `t0`.`published_at` is not null)

I don't think too much about why two queries are run, it was only one in earlier version I tested.

Logically, when utilizing consistent sorting criteria, an item should appear only once across paginated results. While PostgreSQL does not inherently resolve ties in sorting, it falls to Strapi to implement a tie-breaking mechanism within its data retrieval hierarchy to ensure consistent and non-repeating pagination.

**Data:**
name | county
-- | --
Alameda Unified | Alameda
Albany City Unified | Alameda
Berkeley Unified | Alameda
Castro Valley Unified | Alameda
Dublin Unified | Alameda
Fremont Unified | Alameda
Hayward Unified | Alameda
Oakland Unified | Alameda
Piedmont City Unified | Alameda
Pleasanton Unified | Alameda
San Leandro Unified | Alameda
San Lorenzo Unified | Alameda
Amador County Unified | Amador
Biggs Unified | Butte
Chico Unified | Butte
Gridley Unified | Butte
Vallecito Union | Calaveras
Colusa Unified | Colusa
Acalanes Union High | Contra Costa
Antioch Unified | Contra Costa
Brentwood Union Elementary | Contra Costa
Canyon Elementary | Contra Costa
John Swett Unified | Contra Costa
Lafayette Elementary | Contra Costa
Mt. Diablo Unified | Contra Costa
San Ramon Valley Unified | Contra Costa
Walnut Creek Elementary | Contra Costa
West Contra Costa Unified | Contra Costa
Del Norte County Unified | Del Norte
Black Oak Mine Unified | El Dorado
Camino Union Elementary | El Dorado
El Dorado Union High | El Dorado
Lake Tahoe Unified | El Dorado
Placerville Union Elementary | El Dorado
Central Unified | Fresno
Clovis Unified | Fresno
Coalinga-Huron Unified | Fresno
Fowler Unified | Fresno
Fresno Unified | Fresno
Golden Plains Unified | Fresno
Kerman Unified | Fresno
Kings Canyon Joint Unified | Fresno
Pacific Union Elementary | Fresno
Riverdale Joint Unified | Fresno
Capay Joint Union Elementary | Glenn
Hamilton Unified | Glenn
Princeton Joint Unified | Glenn
Big Lagoon Union Elementary | Humboldt
Blue Lake Union Elementary | Humboldt
Eureka City Schools | Humboldt
Fortuna Elementary | Humboldt
Fortuna Union High | Humboldt
Peninsula Union | Humboldt
Rio Dell Elementary | Humboldt
Southern Humboldt Joint Unified | Humboldt
Brawley Elementary | Imperial
Central Union High | Imperial
Holtville Unified | Imperial
Imperial Unified | Imperial
San Pasqual Valley Unified | Imperial
Big Pine Unified | Inyo

**Comment by frederichoule** [comment]
If Strapi Cloud is using PostgreSQL, I have the same bug.

**Comment by muhsinaltintop** [comment]
I am experiencing a similar issue with Strapi version 5.0.2. When I create an entry through the Content Manager, two entries are being created in the database. I noticed that pressing "Save" creates one entry, and pressing "Publish" creates an additional entry. If I directly press "Publish" (as it seems to handle save + publish simultaneously), two entries are created in the database again. However, only one entry is visible in the Strapi panel, and the API also only returns one entry. Due to the presence of two entries in the database, the IDs are incrementing by two.

**Comment by shinchannn** [comment]
> I am experiencing a similar issue with Strapi version 5.0.2. When I create an entry through the Content Manager, two entries are being created in the database. I noticed that pressing "Save" creates one entry, and pressing "Publish" creates an additional entry. If I directly press "Publish" (as it seems to handle save + publish simultaneously), two entries are created in the database again. However, only one entry is visible in the Strapi panel, and the API also only returns one entry. Due to the presence of two entries in the database, the IDs are incrementing by two.

Hi @muhsinaltintop,
It seems to me this design is how they manage content, and it’s likely not related to the issue I’m experiencing. I haven’t checked myself, but if you look at the two records created in the database after hitting publish, one should have a null `published_at` field while the other doesn’t. As shown in the two queries I posted earlier, they first use entries with a null `published_at` field to retrieve the IDs, and then use those IDs to fetch the actual entries. It appears that Strapi is intentionally handling it this way for a specific reason.

**Comment by Marc-Roig** [comment]
This is an ongoing issue we have discussed here https://github.com/strapi/strapi/issues/15953

This could temporary be fixed if you add a second sort by id, unfortunately from the CM we still havent found a proper solution.

Every database has a different behaviour and this is particularly annoying with postgres.  For now I am closing this one in favor of the one I reference dto keep track of the original conversation.

We are open for any ideas that could help fixing this!

**Comment by scale69** [comment]
I also encountered the same issue when fetching data from users with related data. However, I resolved it by displaying only the data that has been published. This approach might differ depending on the case and the implementation.

For example: if publishedAt == null, then do not show the data.

I hope this helps until the bug is fixed.

**Comment by jawor90** [comment]
I can confirm issue still exist in 5.9.0 and PostgreSQL. Fetching entries through API makes duplicate of previous page last element into first element of the next page. 

@Marc-Roig any plans of fixing that?
